jbpm-designer-client/src/main/java/org/jbpm/designer/client/util/AbstractValidatingTextBox.java

```diff
@@ -70,11 +68,13 @@ public void onKeyPress( KeyPressEvent event ) {
                 String newValue = oldValue.substring(0, me.getCursorPos());
                 newValue = newValue + ((char) charCode);
                 newValue = newValue + oldValue.substring(me.getCursorPos() + me.getSelectionLength());
-                if (isValidValue(newValue, false) != null ) {
+                String validationError = isValidValue(newValue, false);
+                if (validationError != null) {
                     event.preventDefault();
+                    fireValidationError(validationError);
```

Comment:
Adding the call to  'fireValidationError' inside onKeyPress works fine, but I think we should use a different error message from the one used when fireValidationError is called from within the BlurHandler. The error message "Removed invalid characters from name: #" makes sense from the BlurHandler, as it's called after the user has pasted an invalid name, e.g. "ab#cd" but from onKeyPress, it would be better to use something like "Invalid character in name: #" (where '#' is the key pressed).
Will you either change this, or creating a  BZ for this change to be made would also be fine. Thanks.